core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java

```diff
@@ -197,36 +197,34 @@ public synchronized long getReservedBytes()
     @NotThreadSafe
     private class TaskDescriptors
     {
-        private final Map<TaskDescriptorKey, TaskDescriptor> descriptors = new HashMap<>();
+        private final Table<StageId, Integer, TaskDescriptor> descriptors = HashBasedTable.create();
+
         private long reservedBytes;
         private RuntimeException failure;
 
         public void put(StageId stageId, int partitionId, TaskDescriptor descriptor)
         {
             throwIfFailed();
-            TaskDescriptorKey key = new TaskDescriptorKey(stageId, partitionId);
-            checkState(descriptors.putIfAbsent(key, descriptor) == null, "task descriptor is already present for key %s ", key);
+            checkState(descriptors.put(stageId, partitionId, descriptor) == null, "task descriptor is already present for key %s/%s ", stageId, partitionId);
```

Comment:
would be nice to have putIfAbsent semantics not to currpt state if incorrect use

maybe descriptors.rowMap  putifabsent